tests/ci/fast_test_check.py

```diff
@@ -90,12 +90,31 @@ def process_results(result_directory: Path) -> Tuple[str, str, TestResults]:
 
     return state, description, test_results
 
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        description="FastTest script",
+    )
+
+    parser.add_argument(
+        "--timeout",
+        type=int,
+        # Fast tests in most cases done within 10 min and 40 min timout should be sufficient,
+        # though due to cold cache build time can be much longer
+        # https://pastila.nl/?146195b6/9bb99293535e3817a9ea82c3f0f7538d.link#5xtClOjkaPLEjSuZ92L2/g==
+        default=40,
+        help="Timeout in minutes",
+    )
+
+    return parser.parse_args()
 
 def main():
     logging.basicConfig(level=logging.INFO)
-
     stopwatch = Stopwatch()
 
+    args = parse_args()
+    timeout = args.timeout * 60
```

Comment:
It feels like the function is a better place for this. Tuning arguments are usually done in `parse_args`

Reply:
done